ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/akkastreams/dispatcher/SignalDispatcher.scala

```diff
@@ -63,14 +63,17 @@ class SignalDispatcher private () extends AutoCloseable {
   /** Closes this SignalDispatcher.
     * For any downstream with pending signals, at least one such signal will be sent first.
     */
-  def close(): Unit =
+  def shutdown(): Future[Unit] = {
+    implicit val ec: ExecutionContext = scala.concurrent.ExecutionContext.parasitic
     runningState
       .getAndSet(None)
-      // note, Materializer's lifecycle is managed outside of this class
-      // fire and forget complete signals -- we can't control how long downstream takes
-      .fold(throw new IllegalStateException("SignalDispatcher is already closed"))(
-        _.foreach(_.complete())
-      )
+      .fold(throw new IllegalStateException("SignalDispatcher is already closed")) { sources =>
+        sources.foreach(_.complete())
+        Future
```

Comment:
So, in case of hitting the timeout this future will live on in the background until finally the sources cave in.
I am not sure what the consequences of that will be other than prohibiting the ref-count on those sources to go to zero (which would happen anyway since they are refusing to die)

Reply:
Me neither. It would be very much the same as original fire and forget behavior. I do not see a way around that (if we want to implement graceful shutdown for API streams). I would guess as the actorsystem is going down, those streams will die as well.

Reply:
:+1: